shared/src/business/entities/cases/CaseExternal.js

```diff
@@ -22,12 +27,13 @@ const { ContactFactory } = require('../contacts/ContactFactory');
  * @param {object} rawCase the raw case data
  * @constructor
  */
-function CaseExternal(rawCase, { applicationContext }) {
-  CaseExternal.prototype.init.call(this, rawCase, { applicationContext });
+function CaseExternal() {}
+CaseExternal.prototype.init = function init(rawCase, { applicationContext }) {
+  CaseExternal.prototype.initSelf.call(this, rawCase, { applicationContext });
```

Comment:
Is there a reason this is preferable to what existed previously? I'm ignorant of most Javascript-related best practices, so I'm asking out of curiosity, and also because this made me a bit cross-eyed to read. 😆

Reply:
The reason for this change was to create a decorator for instantiating the entity with additional validation / sanitation options via joi.

Reply:
Gotcha, I see that now that I've looked at all the code changes. Thanks!